File: Agentic-AI-Travel-Advisor/mobile-app/lib/models/hotel.dart
import 'room.dart';

class Hotel {
  final int id;
  final String name;
  final String address;
  final String city;
  final String country;
  final String? description;
  final int roomCount;
  final List<Room>? rooms;

  const Hotel({
    required this.id,
    required this.name,
    required this.address,
    required this.city,
    required this.country,
    this.description,
    required this.roomCount,
    this.rooms,
  });

  factory Hotel.fromJson(Map<String, dynamic> json) {
    return Hotel(
      id: json['id'] as int,
      name: json['name'] as String,
      address: json['address'] as String,
      city: json['city'] as String,
      country: json['country'] as String,
      description: json['description'] as String?,
      roomCount: json['roomCount'] as int? ?? 0,
      rooms: json['rooms'] != null
          ? (json['rooms'] as List).map((r) => Room.fromJson(r)).toList()
          : null,
    );
  }
}

File: Agentic-AI-Travel-Advisor/shared/Core/Entities/Hotel.cs
using TravelAdvisor.Core.Enums;

namespace TravelAdvisor.Core.Entities;

public class Hotel
{
    public int Id { get; set; }
    public string OwnerId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
    public string? Description { get; set; }
    public ApprovalStatus ApprovalStatus { get; set; } = ApprovalStatus.Pending;

    public ApplicationUser Owner { get; set; } = null!;
    public ICollection<Room> Rooms { get; set; } = [];
}
